src/rgw/driver/rados/rgw_zone.cc

```diff
@@ -1114,6 +1114,11 @@ static int read_or_create_default_zone(const DoutPrefixProvider* dpp,
       return r;
     }
   }
+  r = info.sal_config.set("type", "rados");
```

Comment:
@cbodley @dang when/where is it that you envisioned the default backend to be rados? Is it in the scenario that no json config is provided or if that json is missing a store? I put this here as a place holder for that discussion.

Reply:
My thought is this, and I don't know how hard it is:
The ceph.conf chooses a config provider.  If this is not given, it should default to the RADOS config provider.

Each config provider has a default json if none is stored.  It should probably be the correct store for that provider (so the RADOS provider will default to radosstore, the DB provider should default to DBStore, the json file provider should default to POSIXStore).  If invalid json exists, it should error out.

What do people think about this?

Reply:
@dang agreed. when a `ConfigStore` backend's `create_zone()` function is called with an empty `sal_config`, it should initialize that for the same kind of store. `read_or_create_default_zone()` is generic to all `ConfigStore`s, so the default logic doesn't belong here